### ISSUE TITLE
OpenURI portal dialog lives forever when canceled

### ISSUE THREAD

**Issue by mcatanzaro** [flatpak/xdg-desktop-portal-gtk#140]
Open an OpenURI portal dialog somehow (e.g. by navigating to an ftp:// URI in Epiphany).

Then click the Cancel button.

Expected behavior: the dialog disappears

Actual behavior: the dialog remains displayed forever, with no way to close it or move it. Even 'killall xdg-desktop-portal-gtk' does not work. Only solution is to log out of gnome-shell.

Previous actual behavior: mutter used to crash when clicking Cancel, so I guess the current behavior is a step up from that

**Comment by mcatanzaro** [comment]
Reproducer: open

https://gitlab.gnome.org/GNOME/gnome-settings-daemon/tree/master/subprojects

And click on the gvc link.


BTW I found a workaround. After clicking Cancel, switch to a different workspace, then switch back. Then the dialog will disappear. You can use Epiphany again after closing and reopening it.

**Comment by matthiasclasen** [comment]
I can't reproduce. For me, epiphany crashes before the app chooser comes up

**Comment by mcatanzaro** [comment]
I can only investigate the Epiphany crash if you provide a backtrace.

Anyway, so should be able to reproduce without using Epiphany. Any instance of the OpenURI portal should trigger this issue.

**Comment by mcatanzaro** [comment]
> Reproducer: open
> 
> https://gitlab.gnome.org/GNOME/gnome-settings-daemon/tree/master/subprojects
> 
> And click on the gvc link.

This reproducer no longer works anyway, maybe due to the recent GitLab upgrade on gitlab.gnome.org? It no longer tries to open any weird URI.

I tried triggering the portal manually. TingPing suggested I try `gio open` inside the flatpak runtime. The bug no longer occurs: Cancel works just fine. Perhaps it was fixed by 1.0? Or by some other update....

Anyway, closing this unless I can reproduce it again.
